shell/components/form/LabeledSelect.vue

```diff
@@ -159,6 +148,13 @@ export default {
         return;
       }
 
+      // Force to update the option label if prop has been changed
+      const isOutdated = !this.options.find(({ label }) => option.label === label);
+
+      if (isOutdated) {
+        return this.options.find(({ value }) => option.value === value)?.label;
```

Comment:
This should account for `this.localizedLabel` as lines 161-165 do. It should also reference `this.optionLabel` and `this.optionKey` (if defined)